src/test/java/tatracker/model/group/GroupTest.java

```diff
@@ -0,0 +1,178 @@
+
+    @Test
+    public void sortStudentsAlphabetically() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsAlphabetically();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ALICE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.DANIEL);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.GEORGE);
+    }
+
+    @Test
+    public void sortStudentsByRatingAscending() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByRatingAscending();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ALICE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.BENSON);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.ELLE);
+    }
+
+    @Test
+    public void sortStudentsByRatingDescending() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByRatingDescending();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ELLE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.BENSON);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void sortStudentsByMatricNumber() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByMatricNumber();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.BENSON);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.ELLE);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void hasStudent() {
+        //First student in student list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Middle of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ELLE));
+
+        //End of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.DANIEL));
+
+        //Not in the list
+        assertFalse(MANY_STUDENTS.hasStudent(TypicalStudents.BOB));
+
+        //List has no students
+        assertFalse(NO_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Only one in the list
+        assertTrue(ONE_STUDENT.hasStudent(TypicalStudents.ALICE));
+
+        //Not in the list
+        assertFalse(ONE_STUDENT.hasStudent(TypicalStudents.BOB));
+    }
+
+    @Test
+    public void getIdentifier() {
+        assertEquals("G03", ONE_STUDENT.getIdentifier());
+    }
+
+    @Test
+    public void getGroupType() {
+        assertEquals(GroupType.LAB, ONE_STUDENT.getGroupType());
+        assertEquals(GroupType.TUTORIAL, NO_STUDENTS.getGroupType());
+        assertEquals(GroupType.OTHER, MANY_STUDENTS.getGroupType());
+    }
+
+    @Test
+    public void getStudent() {
+        assertEquals(TypicalStudents.DANIEL, MANY_STUDENTS.getStudent(new Matric("A0192154M")));
+    }
+
+    @Test
+    public void addStudent() {
```

Comment:
Add duplicate student -> throws exception
add null -> throws exception

Reply:
That doesn't go here. It should be tested in UniqueStudentList test which I'm not in charge of.